shub/settings.py

```diff
@@ -0,0 +1,669 @@
+    # Can leave blank if anonymous binding is sufficient
+    AUTH_LDAP_BIND_DN = cfg.AUTH_LDAP_BIND_DN or ""
+    AUTH_LDAP_BIND_PASSWORD = cfg.AUTH_LDAP_BIND_PASSWORD or ""
+
+    # Any user account that has valid auth credentials can login
+    AUTH_LDAP_USER_SEARCH = LDAPSearch(
+        cfg.AUTH_LDAP_USER_SEARCH, ldap.SCOPE_SUBTREE, "(uid=%(user)s)"
+    )
+    AUTH_LDAP_GROUP_SEARCH = LDAPSearch(
+        cfg.AUTH_LDAP_GROUP_SEARCH, ldap.SCOPE_SUBTREE, "(objectClass=posixGroup)"
+    )
+    AUTH_LDAP_GROUP_TYPE = PosixGroupType()
+
+    # Populate the Django user model from the LDAP directory.
+    AUTH_LDAP_USER_ATTR_MAP = {
+        "first_name": "givenName",
+        "last_name": "sn",
+        "email": "mail",
+    }
+
+    # Map LDAP group membership into Django admin flags
+    AUTH_LDAP_USER_FLAGS_BY_GROUP = {
+        # Anyone in this group can get a token to manage images, not superuser
+        "is_staff": cfg.AUTH_LDAP_STAFF_GROUP_FLAGS,
+        # Anyone in this group is a superuser for the app
+        "is_superuser": cfg.AUTH_LDAP_SUPERUSER_GROUP_FLAGS,
+    }
+
+
+# If google_build in use, we are required to include GitHub
+if "google_build" in PLUGINS_ENABLED:
+
+    # For task discovery by celery
+    SOCIAL_AUTH_GITHUB_SCOPE = [
+        "admin:repo_hook",
+        "repo:status",
+        "user:email",
+        "read:org",
+        "admin:org_hook",
+        "deployment_status",
+    ]
+    ENABLE_GITHUB_AUTH = True
+
+# Apply any plugin settings
+for plugin in PLUGINS_ENABLED:
+
+    plugin_module = "shub.plugins." + plugin
+    plugin = import_module(plugin_module)
+
+    # Add the plugin to INSTALLED APPS
+    INSTALLED_APPS.append(plugin_module)
+
+    # Add AUTHENTICATION_BACKENDS if defined, for authentication plugins
+    if hasattr(plugin, "AUTHENTICATION_BACKENDS"):
+        AUTHENTICATION_BACKENDS = (
+            AUTHENTICATION_BACKENDS + plugin.AUTHENTICATION_BACKENDS
+        )
+
+    # Add custom context processors, if defines for plugin
+    if hasattr(plugin, "CONTEXT_PROCESSORS"):
+        for context_processor in plugin.CONTEXT_PROCESSORS:
+            TEMPLATES[0]["OPTIONS"]["context_processors"].append(context_processor)
+
+# Finally, ensure all variables in cfg are set in locals
+for key, value in cfg:
+
+    # Don't set if the value is empty, or it's been set previously
+    if value is None or key in locals() and locals()[key] is not None:
+        continue
+    locals()[key] = value
+
+# Try reading in from secrets first (no issue if not found)
+try:
+    from .secrets import *
+except ImportError:
+    pass
+
+if not MINIO_ROOT_USER or not MINIO_ROOT_PASSWORD:
+    print(
+        "Warning: either MINIO_ROOT_USER or MINIO_ROOT_PASSWORD is missing, storage may not work."
+    )
+
+# If we don't have a secret key, no go
+if "SECRET_KEY" not in locals():
+    sys.exit("SECRET_KEY is required but not set. Set SREGISTRY_SECRET_KEY.")
```

Comment:
Setting `SECRET_KEY` in the YAML file works fantastically, but setting `SREGISTRY_SECRET_KEY` env var doesn't work because we don't define it in the `STRING_DEFAULTS`. It's probably easiest to keep the functionality the way it is now and just update the message to `Set SECRET_KEY in secrets.py or a YAML config file.`

Reply:
Any reason to not allow this in the environment? I am thinking we can just add to the `STRING_DEFAULTS` and be more consistent?

Reply:
No reason I can think of. Adding it to `STRING_DEFAULTS` and updating the check for it to be `not None` makes sense.